src/bg_update.c

```diff
@@ -0,0 +1,561 @@
+                            // _080030DC
+                            u32 index = ((bg->unk20 + r5) - 1);
+                            u16 *r1Ptr = (u16 *)&((u8 *)bg->unk10)[r4 * index * sp08];
+                            u32 index2 = ((bg->unk1E + bg->unk26) - 1);
+                            u16 *r4Ptr = &r1Ptr[index2 * sp08];
+
+                            while (--r5 != (u16)-1) {
+                                // _08003108
+                                u16 i; // <- r3
+                                for (i = 0; i < bg->unk26; i++) {
+                                    // _08003126
+                                    sp3C = &r7Ptr[i];
+                                    *sp3C = (r4Ptr[0 - i] ^ TileMask_FlipXY);
+                                }
+                            }
+                        } else {
+                            // _08003158
+                            u32 index = bg->unk20;
+                            u16 *r1Ptr = (u16 *)&((u8 *)bg->unk10)[r4 * index * sp08];
+                            u32 index2 = bg->unk1E + bg->unk26 - 1;
+                            u16 *r4Ptr = &r1Ptr[index2 * sp08];
+
+                            while (--r5 != (u16)-1) {
+                                // _08003180
+                                u16 i;
+                                for (i = 0; i < bg->unk26; i++) {
+                                    // _0800319E
+                                    sp3C = &r7Ptr[i];
+                                    *sp3C = (r4Ptr[0 - i] ^ TileMask_FlipX);
+                                }
+                            }
+                        }
+                    } else {
+                        // _080031D0
+                        if (bg->unk2E & 0x80) {
+                            // _080031D8
+                            u32 index = ((sp14 + r5) - 1);
+                            u16 *r0Ptr = (u16 *)&((u8 *)bg->unk10)[r4 * index * sp08];
+                            u32 index2 = sp10;
+                            u16 *r4Ptr = &r0Ptr[index2 * sp08];
+
+                            while (--r5 != (u16)-1) {
+                                u16 *r2Ptr = (u16 *)&((u8 *)r7Ptr)[sp0C];
+                                u16 i;
+
+                                for (i = 0; i < bg->unk26; i++) {
+                                    sp3C = &r7Ptr[i];
+                                    *sp3C = r4Ptr[i] ^ TileMask_FlipY;
+                                }
+                                r7Ptr = r2Ptr;
+                            }
+                        } else {
+                            // _08003254
+                            u32 index = sp14;
+                            u16 *r0Ptr = (u16 *)&((u8 *)bg->unk10)[r4 * index * sp08];
+                            u32 index2 = sp10;
+                            u16 *r4Ptr = &r0Ptr[index2 * sp08];
+
+                            // _08003298
+                            while (--r5 != (u16)-1) {
```

Comment:
Are these `CopyLarge`?

Reply:
I don't see a macro that's called that

Reply:
https://github.com/freshollie/sa2/blob/main/include/gba/macro.h#L87

it's probably not this, but maybe in other cases where we are doing `DmaCopy` in loops